File: 0001-two-sum/0001-two-sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for i in range(len(nums)):
            hash_map[nums[i]] = i
        for idx in range(len(nums)):
            complement = target - nums[idx]
            if ((complement) in hash_map) and idx != hash_map[complement]:
                return [idx,hash_map[complement]]
        return - 1
        

File: 0001-two-sum/0001-two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap <Integer, Integer> complementmap = new HashMap <>();
        int complement = 0;
        for(int i = 0; i < nums.length; i ++)
        {
            complement = target - nums[i];
            if (complementmap.containsKey(complement))
            {
               return new int [] {complementmap.get(complement), i};
            }
            else
            {
                complementmap.put(nums[i], i);
            }
        } 

        return new int [0];
    }

}